src/huggingface_hub/utils/_runtime.py

```diff
@@ -177,3 +179,56 @@ def is_google_colab() -> bool:
     Taken from https://stackoverflow.com/a/63519730.
     """
     return _is_google_colab
+
+
+def dump_environment_info() -> Dict[str, str]:
+    from huggingface_hub import HfFolder, whoami
+    from huggingface_hub.utils import list_credential_helpers
+
+    token = HfFolder().get_token()
+
+    # Generic machine info
+    info = {
+        "huggingface_hub version": get_hf_hub_version(),
+        "Platform": platform.platform(),
+        "Python version": get_python_version(),
+    }
+
+    # Interpreter info
+    try:
+        shell_class = get_ipython().__class__  # type: ignore # noqa: F821
+        info["Running in iPython ?"] = "Yes"
+        info["iPython shell"] = shell_class.__name__
+    except NameError:
+        info["Running in iPython ?"] = "No"
+    info["Running in notebook ?"] = is_notebook()
+    info["Running in Google Colab ?"] = is_google_colab()
+
+    # Login info
+    info["Token path ?"] = HfFolder().path_token
+    info["Has saved token ?"] = token is not None
+    if token is not None:
+        info["Token type"] = (
+            "Organization token" if token.startswith("hf_org") else "Personal token"
```

Comment:
I don't think we can ever have an organization token saved right? The login fails for those (see https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/_login.py#L225 )

Reply:
Yep true, I'll remove it. Thanks :)

Reply:
yes i was going to comment the same. Org tokens are on their way to deprecation anyways